src/index/import/markdown.ts

```diff
@@ -43,14 +43,22 @@ export function markdownImport(
     metadata: CachedMetadata,
     stats: FileStats
 ): JsonMarkdownPage {
-    const lines = markdown.split("\n");
-    const frontmatter: Record<string, JsonFrontmatterEntry> | undefined = metadata.frontmatter
-        ? parseFrontmatterBlock(metadata.frontmatter)
-        : undefined;
-
+    
+		const {lines, metadata: pageMetadata, frontmatter, sections} = markdownSourceImport(path, markdown, metadata)
     const page = new PageData(path, stats, lines.length, frontmatter);
+		page.metadata = pageMetadata
+		sections.forEach(s => page.section(s))
 
-    //////////////
+    return page.build();
+}
+export function markdownSourceImport(path: string, markdown: string, metadata: CachedMetadata) {
```

Comment:
This function should be annotated with what type it returns (looks like a map of the major metadata types?).